packages/react/src/modules/core/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+interface SwitchProps {
+  isChecked?: boolean;
+  onColor?: string;
+  onChange?: () => void;
+}
+
+export const Switch: React.FC<SwitchProps> = ({
+  isChecked = false,
+  onColor,
+  onChange,
+}) => {
+  console.log("is Checked", isChecked)
+  const [on, setOn] = useState(isChecked || false);
+
+  const toggleSwitch = () => {
+    if (onChange) {
+      onChange();
+      setOn(!on);
+    } else {
+      setOn(!on);
+    }
+  };
+
+  return (
+    <label className="switch">
+      <input type="checkbox" checked={on} onChange={toggleSwitch} />
+      <span
+        className="slider round"
```

Comment:
I'm not sure the correct answer here, but I think we need some standards around class naming to avoid colliding with end user styles. Maybe prefixing everything with `rnf-` as a lot of components already do.